test/test_foreach.py

```diff
@@ -77,16 +54,11 @@ def __call__(self, inputs, is_cuda, is_fastpath, **kwargs):
             torch.autograd.kineto_available() and
             torch.profiler.ProfilerActivity.CUDA in torch.profiler.supported_activities()
         ):
-            with torch.profiler.profile(activities=(torch.profiler.ProfilerActivity.CPU,)) as p:
+            with torch.profiler.profile() as p:
                 actual = self.func(*inputs, **kwargs)
-            for e in p.key_averages():
-                if e.key == 'cudaLaunchKernel':
-                    if is_fastpath:
-                        assert e.count == self.n_expected_cudaLaunchKernels, \
-                            f'{e.count} != {self.n_expected_cudaLaunchKernels}'
-                    else:
-                        assert e.count > self.n_expected_cudaLaunchKernels, \
-                            f'{e.count} <= {self.n_expected_cudaLaunchKernels}'
+            keys = tuple([e.key for e in p.key_averages()])
+            mta_called = any("multi_tensor_apply_kernel" in k for k in keys)
```

Comment:
I wasn't sure how I can check if `multi_tensor_apply_kernel` is called using `__torch_dispatch__` or `TorchDispatchMode` rather than kineto profiler